### ISSUE TITLE
Accuracy of the RT-DETR without pretrained backbone does not reach the level of the paper

### ISSUE THREAD

**Issue by beauty-girl-cxy** [ultralytics/ultralytics#6924]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi, appreciate your work! When I want to try RT-DETR in this repository，I found the accuracy of the RT-DETR without pre-trained backbone does not reach the level of its paper. Have you encountered similar problems?

And, at first I suspected that it was a problem of not using pre-trained backbone, then I initialize the parameters of the backbone network using the weight parameters of the pre-trained model of the backbone network provided by the author of RT-DETR,  and modified the BN layer in the network to FrozenBN to be consistent with the pre-trained model, but the loss occurred nan. Do you have any corresponding insights on this issue? Thanks a lot!
```python
class FrozenBatchNorm2d(nn.Module):
    """copy and modified from https://github.com/facebookresearch/detr/blob/master/models/backbone.py
    BatchNorm2d where the batch statistics and the affine parameters are fixed.
    Copy-paste from torchvision.misc.ops with added eps before rqsrt,
    without which any other models than torchvision.models.resnet[18,34,50,101]
    produce nans.
    """
    def __init__(self, num_features, eps=1e-5):
        super(FrozenBatchNorm2d, self).__init__()
        n = num_features
        self.register_buffer("weight", torch.ones(n))
        self.register_buffer("bias", torch.zeros(n))
        self.register_buffer("running_mean", torch.zeros(n))
        self.register_buffer("running_var", torch.ones(n))
        self.eps = eps
        self.num_features = n 

    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                              missing_keys, unexpected_keys, error_msgs):
        num_batches_tracked_key = prefix + 'num_batches_tracked'
        if num_batches_tracked_key in state_dict:
            del state_dict[num_batches_tracked_key]

        super(FrozenBatchNorm2d, self)._load_from_state_dict(
            state_dict, prefix, local_metadata, strict,
            missing_keys, unexpected_keys, error_msgs)

    def forward(self, x):
        # move reshapes to the beginning
        # to make it fuser-friendly
        w = self.weight.reshape(1, -1, 1, 1)
        b = self.bias.reshape(1, -1, 1, 1)
        rv = self.running_var.reshape(1, -1, 1, 1)
        rm = self.running_mean.reshape(1, -1, 1, 1)
        scale = w * (rv + self.eps).rsqrt()
        bias = b - rm * scale
        return x * scale + bias

    def extra_repr(self):
        return (
            "{num_features}, eps={eps}".format(**self.__dict__)
        )
```

```python
def initialize_weights(model):
    """Initialize model weights to random values."""
    for m in model.modules():
        t = type(m)
        if t is nn.Conv2d:
            pass  # nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity='relu')
        elif t is nn.BatchNorm2d:
            m.eps = 1e-3
            m.momentum = 0.03
        elif t in [nn.Hardswish, nn.LeakyReLU, nn.ReLU, nn.ReLU6, nn.SiLU]:
            m.inplace = True

    res50 = torch.load('/workspace/ultralytics/ResNet50_vd_ssld_v2_pretrained_from_paddle.pth')
    backbone_dict = dict()
    if 'RTDETRDetectionModel' in str(type(model)):
        # 用rt-detr提供的resnet50预训练权重初始化主干网络：
        new_model_dict = model.state_dict()
        model_keys = list(new_model_dict.keys())
        num = 0
        for key, value in res50.items():
            backbone_dict[model_keys[num]] = value
            num += 1
        new_model_dict.update(backbone_dict)
        model.load_state_dict(new_model_dict)
    
    print('debug')
```

Likewise, to be consistent with the pre-trained model, I also changed the first layer of the backbone from the original one-layer conv to a three-layer conv.
```python
class ResNetLayer(nn.Module):
    """ResNet layer with multiple ResNet blocks."""

    def __init__(self, c1, c2, s=1, is_first=False, n=1, e=4):
        """Initializes the ResNetLayer given arguments."""
        super().__init__()
        self.is_first = is_first

        if self.is_first:
            # self.layer = nn.Sequential(Conv(c1, c2, k=7, s=2, p=3, act=True),
            #                            nn.MaxPool2d(kernel_size=3, stride=2, padding=1))
            self.layer = nn.Sequential(Conv(c1, c2//2, k=3, s=2, p=1, act=True),
                                       Conv(c2//2, c2//2, k=3, s=1, p=1, act=True),
                                       Conv(c2//2, c2, k=3, s=1, p=1, act=True),
                                       nn.MaxPool2d(kernel_size=3, stride=2, padding=1))
        else:
            blocks = [ResNetBlock(c1, c2, s, e=e)]
            blocks.extend([ResNetBlock(e * c2, c2, 1, e=e) for _ in range(n - 1)])
            self.layer = nn.Sequential(*blocks)

    def forward(self, x):
        """Forward pass through the ResNet layer."""
        return self.layer(x)
```
